### DISCUSSION TITLE
Command API?

### DISCUSSION THREAD

**Discussion by Freed-Wu** [SuperBo/fugit2.nvim#98] [Ideas]
Many neovim users are familiar to CLI of git :smile: Can this plugin have a CLI interface? If so, user just call `:Fugit2 add -A` to do something like `git add -A`, not open a window and press `s` to stage all changes. Many vim plugins for git have same design, such as https://github.com/tpope/vim-fugitive 's `:Git XXX` and https://github.com/lambdalisue/vim-gina 's `:Gina XXX`.

**Comment by SuperBo** [comment]
Hi @Freed-Wu, can I have some question about your usage with :Git XXX. What are you unhappy with Fugitive :Git XXX. 

If I implement this feature in Fugit2, what I need to do it different than the Fugitive way?

**Reply by Freed-Wu** [reply]
> What are you unhappy with Fugitive :Git XXX.

Oh, I am just prefer your solution. A lua binding should be better than simply generate a subprocess of git. In fact, it should be the advantage of neovim plugin written in lua: we can do many things different from traditional vim plugin. traditional vim plugin use `system()` to call subprocess, or use `dlopen()` to call a dynamic link library. However, neovim plugin allow developer use lua binding! So I support it.

> If I implement this feature

I think it is simliar to `git` CLI to reduce user's learning cost. in shell we `git XXX YYY`, in neovim, we `Fugit2 XXX YYY` :smile:
